src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,174 @@
+            _localizer[key, arguments];
+
+        /// <summary>
+        /// Creates a new <see cref="IStringLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        IStringLocalizer IStringLocalizer.WithCulture([NotNull] CultureInfo culture) => WithCulture(culture);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments)
+        {
+            var stringValue = _localizer[key].Value;
+            var tokens = new List<string>();
+
+            for (int i = 0; i < stringValue.Length; i++)
+            {
+                if (stringValue[i] == '{')
+                {
+                    StringBuilder myToken = new StringBuilder();
+                    myToken.Append(stringValue[i]);
+                    i++;
+
+                    while (i < stringValue.Length)
+                    {
+                        if (stringValue[i] != '}')
+                        {
+                            if (stringValue[i] == '{')
+                            {
+                                throw new Exception("Malformed resource string");
+                            }
+                            else
+                            {
+                                myToken.Append(stringValue[i]);
+                                i++;
+                            }
+                        }
+                        else
+                        {
+                            myToken.Append(stringValue[i]);
+                            tokens.Add(myToken.ToString());
+                            break;
+                        }
+                    }
+                }
+            }
+            // This call joins all tokens and formats them based on arguments and splits them again into arguments. The
+            // idea here is to encode the formatted arguments. For example if resource value has datetime format as
+            // argument ex: {1:yyyy} we are formatting the argument to right value before encoding.
+            arguments = string.Format(string.Join(",", tokens), arguments).Split(',');
```

Comment:
I started with that but problem with doing it on the fly or using a for loop is  - for tokens other than first one (Ex: {2:x} - you will have `string.Format({2:x}, 100)` which gives an error `Index (zero based) must be greater than or equal to zero and less than the size of the argument list`. This is because you just have 1 argument in `string.Format` at any point but you are starting it with 2 not 0. So either I have to combine all tokens and split them after with some character so I don't get this error or if I am doing it on the fly, I should replace tokens other than 0 with 0 every time.